documentation/docs/tutorials/single-topic-multi-tenant.mdx

```diff
@@ -0,0 +1,360 @@
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }],
+    "reason": "Add tenant1 data source"
+  }'
+```
+
+### Add additional tenants
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant2",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant2"
+    }],
+    "reason": "Add tenant2 data source"
+  }'
+```
+
+Note: No service restart is required when adding new tenants.
+
+## Verification
+
+### Check data isolation
+
+```bash
+# Tenant 1 data
+curl -s http://localhost:8181/v1/data/acl/tenant1 | jq .
+
+# Tenant 2 data
+curl -s http://localhost:8181/v1/data/acl/tenant2 | jq .
+
+# All tenant data
+curl -s http://localhost:8181/v1/data/acl | jq .
+```
+
+### Monitor OPAL logs
+
+OPAL Server logs:
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml logs -f opal_server
+```
+
+OPAL Client logs:
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml logs -f opal_client
+```
+
+Look for these log patterns:
+
+**Server side:**
+```
+Publishing data update to topics: {'tenant_data'}
+Broadcasting incoming event: {'topic': 'tenant_data'}
+```
+
+**Client side:**
+```
+Received notification of event: tenant_data
+Fetching data from url: http://example_external_data_provider:80/acl/tenant1
+Saving fetched data to policy-store: destination path='/acl/tenant1'
+```
+
+## Data Updates
+
+### Update existing tenant data
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant1",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }],
+    "reason": "Refresh tenant1 data"
+  }'
+```
+
+## Testing Authorization
+
+### Test tenant-specific policies
+
+```bash
+curl -X POST http://localhost:8181/v1/data/policies/rbac/allow \
+  -H "Content-Type: application/json" \
+  -d '{
+    "input": {
+      "user": "alice",
+      "action": "read",
+      "resource": "document1",
+      "tenant_id": "tenant1"
+    }
+  }' | jq .
+```
+
+## Production Considerations
+
+### Use Policy Repo webhooks instead of polling
+
+For production environments, configure webhooks for immediate policy updates:
+
+```yaml
+opal_server:
+  environment:
+    - OPAL_POLICY_REPO_WEBHOOK_SECRET=your-secure-secret
+```
+
+### Secure data sources
+
+Configure authentication for external data APIs:
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "https://secure-api.yourdomain.com/tenant-data/tenant1",
+      "config": {
+        "headers": {
+          "Authorization": "Bearer your-secure-token"
+        }
+      },
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }]
+  }'
+```
+
+### Scale horizontally
```

Comment:
This is not Horizontal Scaling, it is inside the same pod / server 😬
Adding additional Uvicorn workers can help utilizing the server resources better, useful for Vertical Scaling when increasing the CPU / Memory of the server.